### PR TITLE
GetNode/Segment -> ToNode/Segment

### PR DESCRIPTION
**Pull request by kianzarrin**
GetNode(nodeID) with ToNode()
GetSeg(segmentID) with ToSegment()
renamed:
GetNode(segment, startNode) to GetNodeID(segment, startNode) because the return value is ushort not ref NetNode

### CODE REVIEW

File: TLM/TLM/Manager/Impl/LaneArrowManager.cs [hunk 1]
```diff
@@ -182,11 +182,11 @@ public class LaneArrowManager
         private static void RecalculateFlags(uint laneId) {
             NetLane[] laneBuffer = NetManager.instance.m_lanes.m_buffer;
             ushort segmentId = laneBuffer[laneId].m_segment;
-            NetAI ai = GetSeg(segmentId).Info.m_netAI;
+            NetAI ai = segmentId.ToSegment().Info.m_netAI;
 #if DEBUGFLAGS
             Log._Debug($"Flags.RecalculateFlags: Recalculateing lane arrows of segment {segmentId}.");
 #endif
-            ai.UpdateLanes(segmentId, ref GetSeg(segmentId), true);
+            ai.UpdateLanes(segmentId, ref segmentId.ToSegment(), true);
```
Comment (DaEgi01) [inline TLM/TLM/Manager/Impl/LaneArrowManager.cs:189]: no need to do it twice for the same segment.

File: TLM/TLM/Manager/Impl/LaneConnectionManager.cs [hunk 2]
```diff
@@ -71,7 +71,7 @@ public class LaneConnectionManager
         private bool IsHeadingTowardsStartNode(uint sourceLaneId) {
             NetLane[] laneBuffer = NetManager.instance.m_lanes.m_buffer;
             ushort segmentId = laneBuffer[sourceLaneId].m_segment;
-            NetSegment segment = GetSeg(segmentId);
+            NetSegment segment = segmentId.ToSegment();
```
Comment (DaEgi01) [inline TLM/TLM/Manager/Impl/LaneConnectionManager.cs:74]: ref ref :D

**Review by DaEgi01 — CHANGES_REQUESTED** [review]

File: TLM/TLM/Manager/Impl/LaneArrowManager.cs (hunk 1 above)
Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/LaneArrowManager.cs:189]: this is not performance critical. in any case I am not doing anything the old code didn't do.

File: TLM/TLM/Manager/Impl/LaneConnectionManager.cs (hunk 2 above)
Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/LaneConnectionManager.cs:74]: you should take a look at macsurgeys code! he refuses to use ref as a matter of principle!

**Review by krzychu124 — APPROVED** [review]

File: TLM/TLM/Manager/Impl/LaneArrowManager.cs (hunk 1 above)
Comment (DaEgi01) [reply TLM/TLM/Manager/Impl/LaneArrowManager.cs]: yeah I would have probably done it in a separate commit too

File: TLM/TLM/Manager/Impl/LaneConnectionManager.cs (hunk 2 above)
Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/LaneConnectionManager.cs]: I was just kidding!
@DaEgi01 In any case if you didn't notice I already fixed it.

Comment (kianzarrin) [reply TLM/TLM/Manager/Impl/LaneConnectionManager.cs]: image

**Review by DaEgi01 — APPROVED** [review]

File: TLM/TLM/Manager/Impl/LaneConnectionManager.cs (hunk 2 above)
Comment (DaEgi01) [reply TLM/TLM/Manager/Impl/LaneConnectionManager.cs]: lol :D

**Pull request merged**